src/main/java/org/apache/commons/lang3/BooleanUtils.java

```diff
@@ -738,7 +738,7 @@ public static Boolean toBooleanObject(final String str) {
         // Optimisation provides same performance as before for interned 'true'.
         // Similar performance for null, 'false', and other strings not length 2/3/4.
         // 'true'/'TRUE' match 4 times slower, 'tRUE'/'True' 7 times slower.
-        if (str == TRUE) {
+        if ("TRUE".equals(str)) {
```

Comment:
-1: This makes no sense to me and breaks the optimized code below in the sense that a lot of it would now never be called. Why?